src/ngraph/runtime/cpu/mkldnn_emitter.cpp

```diff
@@ -114,12 +114,9 @@ mkldnn::memory::desc
 
 size_t MKLDNNEmitter::build_memory_primitive(const mkldnn::memory::desc& desc)
 {
-    // The MKL-DNN C++ API forces proper initialization of a memory primitive
-    // with a non-null pointer (unlike the C API)
-    // Primitives are initialized at runtime so we use a known-invalid address here
-    // to bypass this check
-    return insert_primitive(
-        new mkldnn::memory({desc, mkldnn_utils::global_cpu_engine}, reinterpret_cast<void*>(0x42)));
+    size_t index =
+        insert_primitive(new mkldnn::memory({desc, mkldnn_utils::global_cpu_engine}, nullptr));
```

Comment:
I tried this yesterday..apparently I was hitting some memory corruption issue on SKX.

Reply:
checked again. still it crashes for me 
```(.venv_mxnet) [pthoreho@nervana-skx21 build]$ ./test/unit-test --gtest_filter="CPU.convolution*"
Note: Google Test filter = CPU.convolution*
[==========] Running 32 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 32 tests from CPU
[ RUN      ] CPU.convolution_2d_1item
*** Error in `./test/unit-test': corrupted size vs. prev_size: 0x000055f1441445c0 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f8d5dc287e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x7e9dc)[0x7f8d5dc2f9dc]
/lib/x86_64-linux-gnu/libc.so.6(+0x81cde)[0x7f8d5dc32cde]
/lib/x86_64-linux-gnu/libc.so.6(__libc_malloc+0x54)[0x7f8d5dc35184]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_Znwm+0x18)[0x7f8d5e527e78]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_Znam+0x9)[0x7f8d5e527f19]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_ZNSt13basic_filebufIcSt11char_traitsIcEE27_M_allocate_internal_bufferEv+0x28)[0x7f8d5e586468]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_ZNSt13basic_filebufIcSt11char_traitsIcEE4openEPKcSt13_Ios_Openmode+0x62)[0x7f8d5e58a6b2]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_ZNSt14basic_ofstreamIcSt11char_traitsIcEEC1ERKNSt7__cxx1112basic_stringIcS1_SaIcEEESt13_Ios_Openmode+0xc8)[0x7f8d5e58b2a8]
/dataset/ngraph/build/src/ngraph/libcpu_backend.so.0.6(_ZN6ngraph7runtime3cpu16GenerateTimelineERKSt6vectorINS1_12OpAttributesESaIS3_EEPlRKNSt7__cxx1112basic_stringIcSt11char_tra
itsIcESaIcEEE+0x52)[0x7f8d6cb87042]
/dataset/ngraph/build/src/ngraph/libcpu_backend.so.0.6(_ZN6ngraph7runtime3cpu13CPU_CallFrame4callERKSt6vectorISt10shared_ptrINS0_10TensorViewEESaIS6_EESA_+0x33c)[0x7f8d6cb56b2c]
/dataset/ngraph/build/src/ngraph/libcpu_backend.so.0.6(_ZN6ngraph7runtime3cpu11CPU_Backend4callESt10shared_ptrINS_8FunctionEERKSt6vectorIS3_INS0_10TensorViewEESaIS8_EESC_+0xcc)[0
x7f8d6ca4d36c]
./test/unit-test(_ZN6ngraph7runtime7Backend18call_with_validateESt10shared_ptrINS_8FunctionEERKSt6vectorIS2_INS0_10TensorViewEESaIS7_EESB_+0x103)[0x55f1410fdde3]
./test/unit-test(_ZN29CPU_convolution_2d_1item_Test8TestBodyEv+0x6da)[0x55f1411264da]
./test/unit-test(_ZN7testing8internal38HandleSehExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc+0x7a)[0x55f141a9823a]
./test/unit-test(_ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc+0x77)[0x55f141a81667]
./test/unit-test(_ZN7testing4Test3RunEv+0xc5)[0x55f141a65bb5]
./test/unit-test(_ZN7testing8TestInfo3RunEv+0xde)[0x55f141a66b0e]
./test/unit-test(_ZN7testing8TestCase3RunEv+0xe7)[0x55f141a672f7]
```
 on this commit

```commit ebd6904c53b9aca6df7456cc73da8c83a8e7fb07 (HEAD -> pruthvi/mkldnn_memory, origin/pruthvi/mkldnn_memory, origin/master, origin/HEAD, master)
Author: Robert Kimball <robert.kimball@intel.com>
Date:   Thu Aug 23 14:33:02 2018 -0700

    revert exclude from all on directory to get codegen library to install (#1482)```

the only changes I have here is 

```(.venv_mxnet) [pthoreho@nervana-skx21 build]$ git diff
diff --git a/src/ngraph/runtime/cpu/mkldnn_emitter.cpp b/src/ngraph/runtime/cpu/mkldnn_emitter.cpp
index 9121ca21..1c6f483c 100644
--- a/src/ngraph/runtime/cpu/mkldnn_emitter.cpp
+++ b/src/ngraph/runtime/cpu/mkldnn_emitter.cpp
@@ -118,8 +118,7 @@ size_t MKLDNNEmitter::build_memory_primitive(const mkldnn::memory::desc& desc)
     // with a non-null pointer (unlike the C API)
     // Primitives are initialized at runtime so we use a known-invalid address here
     // to bypass this check
-    return insert_primitive(
-        new mkldnn::memory({desc, mkldnn_utils::global_cpu_engine}, reinterpret_cast<void*>(0x42)));
+    return insert_primitive(new mkldnn::memory({desc, mkldnn_utils::global_cpu_engine}, nullptr));
 }
```
don't know if I need reshape changes as well?